frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -410,6 +442,33 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
     >
       {render()}
       <Prompt message={handleLocationChange}></Prompt>
+      <GenericModal
+        message="Are you sure you want to permanently delete the property from inventory?"
+        size={ModalSize.SMALL}
+        cancelButtonText="Cancel"
+        okButtonText="Delete"
+        display={showDelete}
+        setDisplay={setShowDelete}
+        handleOk={async () => {
+          try {
+            switch (context) {
+              case SidebarContextType.UPDATE_BUILDING:
+              case SidebarContextType.VIEW_BUILDING:
+                await deleteBuilding(buildingDetail as IBuilding)(dispatch);
+                break;
+              default:
+                await deleteParcel(parcelDetail as IParcel)(dispatch);
+                break;
+            }
+            setShowSideBar(false, undefined, undefined, true);
+          } catch (error) {
+            debugger;
```

Comment:
We need to handle errors like concurrency, but for some reason the error is always `undefined`.